### DISCUSSION TITLE
Should System Email Address be defined per site?

### DISCUSSION THREAD

**Discussion by Kashkin** [craftcms/cms#8446] [Ideas]
This would allow notification emails for new user accounts to come from an email address that matches the current site.

This is potentially also relevant for Craft Commerce running multiple storefronts across sites.

**Comment by brandonkelly** [comment]
Not a bad idea.

This isn’t elegant, but you can currently pull this off by creating a `config/app.php` file with this:

```php
<?php

return [
    'components' => [
        'mailer' => function() {
            $settings = Craft::$app->getSystemSettings()->getEmailSettings();

            switch (CRAFT_SITE) {
                case 'siteA':
                    $settings->fromEmail = 'hello@site-a.com';
                    break;
                case 'siteB':
                    $settings->fromEmail = 'hello@site-b.com';
                    break;
            }

            return craft\helpers\MailerHelper::createMailer($settings);
        },
    ],
];
```

`$settings` will be a [`craft\models\MailSettings`](https://github.com/craftcms/cms/blob/develop/src/models/MailSettings.php) object. In addition to `fromEmail`, you can also override `fromName` (name of sender), `template` (HTML email template path), `transportType` (name of the transporter class), and `transportSettings` (settings the transporter should be initialized with).
